### PR TITLE
chore: refactor noBundle to make it clearer

### PR DESCRIPTION
**Pull request by rozenmd**

### CODE REVIEW

File: packages/wrangler/src/dev.tsx [hunk 1]
```diff
@@ -497,9 +497,10 @@ export async function startApiDev(args: StartDevOptions) {
 			configParam
 		);
 
+		const disableBundling = configParam.no_bundle ?? !args.bundle;
```
Comment (threepointone) [inline packages/wrangler/src/dev.tsx:500]: args.bundle should override configParam.no_bundle, so this order might be wrong

Comment (rozenmd) [reply packages/wrangler/src/dev.tsx:500]: Gah

File: packages/wrangler/src/api/dev.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ interface DevOptions {
 	env?: string;
 	ip?: string;
 	port?: number;
-	noBundle?: boolean;
```
Comment (threepointone) [inline packages/wrangler/src/api/dev.ts:13]: you changed this here, but not anywhere else it's read. intentional?

Comment (rozenmd) [reply packages/wrangler/src/api/dev.ts:13]: Yeah, we don't actually use `args.noBundle`, it's `args.bundle`

Comment (rozenmd) [reply packages/wrangler/src/api/dev.ts:13]: Incorrect typing

**Review by threepointone — APPROVED** [review]
approving, but I recommend manually testing this as well before landing. I remember it being a pita when I implemented it.

File: packages/wrangler/src/dev.tsx [hunk 3]
```diff
@@ -497,9 +497,13 @@ export async function startApiDev(args: StartDevOptions) {
 			configParam
 		);
 
+		//if args.bundle is on, don't disable bundling
+		//if there's no args.bundle, and configParam.no_bundle is on, disable bundling
+		//otherwise, enable bundling
+		const disableBundling = !(args.bundle ?? !configParam.no_bundle);
```
Comment (petebacondarwin) [inline packages/wrangler/src/dev.tsx:503]: How about:

```suggestion
		const enableBundling = args.bundle ?? !configParam.no_bundle;
```

Then lower down:

```
noBundle: !enableBundling
```

**Review by petebacondarwin — CHANGES_REQUESTED** [review]

**Pull request merged**